src/seedu/addressbook/Main.java

```diff
@@ -103,16 +105,19 @@ private void recordResult(CommandResult result) {
      * @param command user command
      * @return result of the command
      */
-    private CommandResult executeCommand(Command command)  {
+    private CommandResult executeCommand(Command command) throws StorageFileDeletedException {
         try {
             command.setData(addressBook, lastShownList);
             CommandResult result = command.execute();
-            storage.save(addressBook);
+            storage.save(addressBook, true);
             return result;
+        } catch (StorageFileDeletedException e){
```

Comment:
this is acceptable. but i don't understand if you are doing this, why are you still throwing exceptions to higher classes?